include/z64pause.h

```diff
@@ -0,0 +1,137 @@
+#ifndef Z64PAUSE_H
+#define Z64PAUSE_H
+
+#include "z64animation.h"
+#include "z64message.h"
+#include "z64view.h"
+
+struct OcarinaStaff;
+
+typedef enum {
+    /* 0 */ PAUSE_BG_PRERENDER_OFF, // Inactive, do nothing.
+    /* 1 */ PAUSE_BG_PRERENDER_SETUP, // The current frame is only drawn for the purpose of serving as the pause background.
+    /* 2 */ PAUSE_BG_PRERENDER_PROCESS, // The previous frame was PAUSE_BG_PRERENDER_SETUP, now apply prerender filters.
+    /* 3 */ PAUSE_BG_PRERENDER_DONE, // The pause background is ready to be used.
+    /* 4 */ PAUSE_BG_PRERENDER_MAX
+} PauseBgPreRenderState;
```

Comment:
something tells me this sorta belongs to z_play more than anything else, but I understand why it makes sense here too. But the different states are related more to how z_play should processing a drawing frame more than anything 

curious on other thoughts, if the concensus is keep it here then thats fine

Reply:
Yeah I guess the implementation is more z_play-ish than pause-ish, I just grabbed it since "hey it has pause in the name". I'll move it (comments on (not) moving it still welcome)

Reply:
Wait I thought for some reason "z_play" had its own header but it doesn't

so instead of leaving this in z64.h for now I think it's better in z64pause.h

Reply:
Small note, but whatever happens to `PauseBgPreRenderState` the same thing should probably happen to `TransitionTileStatus` (after it gets merged of course :) )

Reply:
> Small note, but whatever happens to `PauseBgPreRenderState` the same thing should probably happen to `TransitionTileStatus` (after it gets merged of course :) )

wdym? Move `TransitionTileStatus` to z64pause.h too if we do leave `PauseBgPreRenderState` in z64pause.h ?
Or move `TransitionTileStatus` to z64.h (with `PlayState`) if we do leave `PauseBgPreRenderState` in z64.h ?

Reply:
If `PauseBgPreRenderState` stays in z64pause.h , then `TransitionTileStatus` should probably stay in z64transition (or possibly `fbdemo/TransitionTile` gets its own header eventually).  If we do leave `PauseBgPreRenderState` in z64.h then `TransitionTileStatus` should probably also go in z64.h with `PlayState`

Reply:
`so instead of leaving this in z64.h for now I think it's better in z64pause.h`
I dont really agree, it staying in z64.h just means it doesnt have a home yet, which is true (until play is properly split)

Reply:
Alrighty I moved both enums (pausebg and transitiontile) near PlayState